### DISCUSSION TITLE
How does kong support sse(Server Sent Events)

### DISCUSSION THREAD

**Discussion by silenceboychen** [Kong/kong#11151] [Help]
I created an sse(Server Sent Events) interface, and everything works fine in the local test, but an error occurs when the publishing server requests through the kong gateway

The error message is as follows

```
Kong Error
An unexpected error occurred.
```

I don't know how to configure

**Comment by fffonion** [comment]
Hi @silenceboychen you might need following directives added to [custom nginx template](https://docs.konghq.com/gateway/latest/reference/nginx-directives/) of kong
```
proxy_cache off;  # Turn off caching
proxy_buffering off;  # Turn off proxy buffering
chunked_transfer_encoding on;  # Turn on chunked transfer encoding
tcp_nopush on;  # Turn on TCP NOPUSH option, disable Nagle algorithm
tcp_nodelay on;  # Turn on TCP NODELAY option, disable delay ACK algorithm
keepalive_timeout 300;  # Set keep-alive timeout to 65 seconds

```

**Reply by silenceboychen** [reply]
thanks, solved this problem

**Reply by ercanucan** [reply]
@silenceboychen could you share how you solved this problem? did you place these values into kong.conf with `nginx_proxy_` prefix like below? thx!

```
nginx_proxy_proxy_cache off;  # Turn off caching
nginx_proxy_proxy_buffering off;  # Turn off proxy buffering
nginx_proxy_chunked_transfer_encoding on;  # Turn on chunked transfer encoding
nginx_proxy_tcp_nopush on;  # Turn on TCP NOPUSH option, disable Nagle algorithm
nginx_proxy_tcp_nodelay on;  # Turn on TCP NODELAY option, disable delay ACK algorithm
nginx_proxy_keepalive_timeout 300;  # Set keep-alive timeout to 65 seconds
```

**Reply by ercanucan** [reply]
In our case we were able to solve it by disabling the response buffering at ingress configration via https://docs.konghq.com/kubernetes-ingress-controller/latest/references/annotations/#konghqcomresponse-buffering

**Reply by silenceboychen** [reply]
@ercanucan I didn't change any configuration of kong. This problem can be solved by upgrading the http request protocol from 1.1 to 2.0

**Reply by chris-aeviator** [reply]
@silenceboychen awesome! could you elaborate on how to do this? When upgrading the request (node-fetch), I still see my connection disconnected after exactly 15 seconds

> 
> const fetch = require('node-fetch')
> const https = require('https');
> const { Agent } = require('https');
> 
> 
> async function readStream(body){
>   const iter = readableStreamAsyncIterable(body);
> 
>   for await (const chunk of iter) {
>     console.log(chunk)
>   }
> }
> 
> 
> function readableStreamAsyncIterable(stream) {
>   if (stream[Symbol.asyncIterator]) return stream;
> 
> }
> 
> async function main(){
>   const http2Agent = new Agent({ keepAlive: true, protocol: 'http2:' });
>   const result = await fetch("https://rag-app.ohmy.tofu/stream",
>      {
>     agent: http2Agent,
>       method: 'GET'
>        }
>   );
>   readStream(result.body)
> }
> main()
> `´```````````

### Update
I realized the issue comes when using a kuma endpoint, when using http 2 and using kong to directly point to my service (instead of via a kuma dp & gw), my connection doesnt get dropped!

**Reply by tarunmittal24** [reply]
i also solved my error, using this:-

https://docs.konghq.com/kubernetes-ingress-controller/latest/references/annotations/#konghqcomresponse-buffering

Thanks @ercanucan
